web/src/components/apps/Dashboard.jsx

```diff
@@ -176,6 +176,14 @@ class Dashboard extends Component {
 
   getAppDashboard = () => {
     return new Promise((resolve, reject) => {
+      // this function is in a repeating callback that terminates when
+      // the promise is resolved
+
+      // TODO: use react-query to refetch this instead of the custom repeater
+      if (!this.props.app) {
```

Comment:
The dashboard is hanging on the react 18 upgrade branch. The data is loading but the loader just stays. I'm not able to recreate it locally so I'm trying out a couple of small tweaks. 

This gives a warning locally because the fetch fails. The fetch fails because it's called before the app state is set. This function is called in a repeater so exiting when the app data is not available with return is fine.